src/server.rs

```diff
@@ -1095,4 +1119,45 @@ mod tests {
         second_socket.shutdown(std::net::Shutdown::Both).unwrap();
         assert!(server.requests().is_ok());
     }
+
+    #[test]
+    fn test_kill_switch() {
+        let path_to_socket = get_temp_socket_file();
+
+        let mut server = HttpServer::new(path_to_socket.as_path()).unwrap();
+        let kill_switch = EventFd::new(EFD_NONBLOCK).unwrap();
+        server
+            .add_kill_switch(kill_switch.try_clone().unwrap())
+            .unwrap();
+        server.start_server().unwrap();
+
+        let request_result = Arc::new(Mutex::new(Ok(vec![])));
+        let res_clone = request_result.clone();
+        // Start a thread running the server, expect it to report shutdown event.
+        let handler = std::thread::spawn(move || {
+            *res_clone.lock().unwrap() = server.requests();
+        });
+
+        // Trigger kill switch.
+        kill_switch.write(1).unwrap();
+        // Then send request.
+        let mut socket = UnixStream::connect(path_to_socket.as_path()).unwrap();
+        socket
+            .write_all(
+                b"PATCH /machine-config HTTP/1.1\r\n\
+                         Content-Length: 13\r\n\
+                         Content-Type: application/json\r\n\r\nwhatever body",
+            )
+            .unwrap();
+        // Wait for server thread to handle events.
+        handler.join().unwrap();
+
+        // Expect shutdown event instead of http request event.
+        match &*request_result.lock().unwrap() {
+            Err(ServerError::ShutdownEvent) => (),
+            v => {
+                panic!("Expected shutdown event, instead got {:?}.", v)
+            }
+        };
```

Comment:
Can `Eq` be implemented on `ServerError` to do this more cleanly?
```suggestion
        let res = request_result.lock().unwrap();
        assert_eq!(
            res, Err(ServerError::ShutdownEvent), 
            "Expected shutdown event, instead got {res:?}"
        );
```
If not I'd suggest
```rust
        let res = request_result.lock().unwrap();
        assert!(matches!(res, Err(ServerError::ShutdownEvent)));
```

Reply:
done :+1: